universal-application-tool-0.0.1/app/services/applicant/ApplicantServiceImpl.java

```diff
@@ -61,4 +99,59 @@ public ApplicantServiceImpl(
             },
             httpExecutionContext.current());
   }
+
+  /**
+   * Tries to perform an in-place update of {@link ApplicantData}. Returns error messages if it is
+   * impossible to perform the operation.
+   */
+  private ImmutableSet<String> stageUpdates(
+      Applicant applicant,
+      ProgramDefinition programDefinition,
+      long blockId,
+      ImmutableSet<Update> updates) {
+
+    Optional<BlockDefinition> blockDefinitionMaybe = programDefinition.getBlockDefinition(blockId);
+    if (blockDefinitionMaybe.isEmpty()) {
+      return ImmutableSet.of("Block not found.");
+    }
+    BlockDefinition blockDefinition = blockDefinitionMaybe.get();
```

Comment:
suggestion (blocking):

```suggestion
    BlockDefinition blockDefinition = programDefinition.getBlockDefinition(blockId).orElseThrow​(() -> new ProgramBlockNotFoundException(programDefinitition.id(), blockId));
```

Reply:
done